### ISSUE TITLE
DataStoreException{message=Error in querying the model >> ambiguous column name: User.id

### ISSUE THREAD

**Issue by iamAliRaza** [aws-amplify/amplify-android#1572]
Hi Guyz, I am facing issue while fetching the list of comments from Comment table. I am not sure but i think it's happening because of model relationships.  **Need help !!**

Look at these 3 models. Comment Model has one to one relation with User. And post also has one to one relation with a user.
Comment also has a one to many relation with Post.

Where i am wrong please let me know??

```
type User @model  @auth(rules: [{ allow: private }]) {
  id: ID!
  email: String!
  username: String
  phoneNumber:String
  sex:String
  state:String
  city:String
  zipcode:String
  interest:String
  status:String
  userImage:String
  professions:[String]
  danceStyles:[String]
}

type Post @model @auth(rules: [{ allow: private }]){
    id:ID!
    postImage:String
    caption:String
    date:String
    startTime:String
    endTime:String
    cost:Int
    paymentLink:String
    location : String
    about : String
    likeList : [String]
    bookmarkList : [String]
    postType : PostType
    categories : [String]
    danceStyles : [String]
    postUser : User @connection
    comments: [Comment] @connection(keyName: "byPost", fields: ["id"])
}


type Comment @model @auth(rules: [{ allow: private }])
@key(name: "byPost", fields: ["postID", "comment"]){
    id : ID!
    comment: String!
    postID: ID!
    post: Post! @connection(fields: ["postID"])
    commentedUser : User @connection
    likeList : [String]
}
```

Error that i am getting (while querying Comment model):

> ambiguous column name: User.id
> 
> while compiling: SELECT `Comment`.`id` AS `Comment_id`, `Comment`.`comment` AS `Comment_comment`, `Comment`.`createdAt` AS `Comment_createdAt`, `Comment`.`likeList` AS `Comment_likeList`, `Comment`.`updatedAt` AS `Comment_updatedAt`, `Comment`.`commentCommentedUserId` AS `Comment_commentCommentedUserId`, `Comment`.`postID` AS `Comment_postID`, `User`.`id` AS `User_id`, `User`.`city` AS `User_city`, `User`.`createdAt` AS `User_createdAt`, `User`.`danceStyles` AS `User_danceStyles`, `User`.`email` AS `User_email`, `User`.`interest` AS `User_interest`, `User`.`phoneNumber` AS `User_phoneNumber`, `User`.`professions` AS `User_professions`, `User`.`sex` AS `User_sex`, `User`.`state` AS `User_state`, `User`.`status` AS `User_status`, `User`.`updatedAt` AS `User_updatedAt`, `User`.`userImage` AS `User_userImage`, `User`.`username` AS `User_username`, `User`.`zipcode` AS `User_zipcode`, `Post`.`id` AS `Post_id`, `Post`.`about` AS `Post_about`, `Post`.`bookmarkList` AS `Post_bookmarkList`, `Post`.`caption` AS `Post_caption`, `Post`.`categories` AS `Post_categories`, `Post`.`cost` AS `Post_cost`, `Post`.`createdAt` AS `Post_createdAt`, `Post`.`danceStyles` AS `Post_danceStyles`, `Post`.`date` AS `Post_date`, `Post`.`endTime` AS `Post_endTime`, `Post`.`likeList` AS `Post_likeList`, `Post`.`location` AS `Post_location`, `Post`.`paymentLink` AS `Post_paymentLink`, `Post`.`postImage` AS `Post_postImage`, `Post`.`postType` AS `Post_postType`, `Post`.`startTime` AS `Post_startTime`, `Post`.`updatedAt` AS `Post_updatedAt`, `Post`.`postPostUserId` AS `Post_postPostUserId`, `User`.`id` AS `User_id`, `User`.`city` AS `User_city`, `User`.`createdAt` AS `User_createdAt`, `User`.`danceStyles` AS `User_danceStyles`, `User`.`email` AS `User_email`, `User`.`interest` AS `User_interest`, `User`.`phoneNumber` AS `User_phoneNumber`, `User`.`professions` AS `User_professions`, `User`.`sex` AS `User_sex`, `User`.`state` AS `User_state`, `User`.`status` AS `User_status`, `User`.`updatedAt` AS `User_updatedAt`, `User`.`userImage` AS `User_userImage`, `User`.`username` AS `User_username`, `User`.`zipcode` AS `User_zipcode` FROM `Comment` LEFT JOIN `User` ON `Comment`.`commentCommentedUserId`=`User`.`id` INNER JOIN `Post` ON `Comment`.`postID`=`Post`.`id`LEFT JOIN `User` ON `Post`.`postPostUserId`=`User`.`id` WHERE `Comment`.`postID` = ?;

**Comment by eeatonaws** [comment]
Hi @iamAliRaza, thanks for reporting this issue. Could you please provide the version of Amplify you're using and a code snippet for your DataStore query?

**Comment by iamAliRaza** [comment]
@eeatonaws  yeah sure ... 

```
Amplify Datastore version : 
implementation 'com.amplifyframework:aws-datastore:1.28.3'
```

Almost all dependencies i am using the latest versions as well as Amplify CLI is also updated.


Queries that i tried and got that error:

```
Amplify.DataStore.query(
            Comment::class.java,
            Where.matches(Comment.POST.eq(postId)),
            { result ->
                
            },
            {
                it.printStackTrace()
            })
            
            
            And
            
            Amplify.DataStore.query(Comment::class.java,{ results ->},
           {            
            it.printStackTrace()
        })
```

**Comment by eeatonaws** [comment]
@iamAliRaza, I'm working on reproducing the issue but in your GraphQL schema, the Post model has `postType : PostType` which is not one of the models in the schema you provided. Could you provide the definition for PostType?

**Comment by iamAliRaza** [comment]
Hi @eeatonaws  I didn't check this behaviour again. I migrated my backend to graphQL v2 transformer and after that everything sudden stopped working because of issues with Amplify Codegen Model command.. Its generating incorrect models if we are using any relation in model. i reported this bug as well and talked with support and ended up Amplify is unstable platform sorry but its a fact. 

Another fact is there is no complete BAAS platform for mobile developers except Firebase.

Check below >> PostType Model:

enum PostType {
    EVENT_POST
    DANCE_CLASS_POST
}

**Comment by eeatonaws** [comment]
A fix for this issue has been released as part of Amplify Android 1.31.1.
